### ISSUE TITLE
Failing test[s] with Emacs 29.1

### ISSUE THREAD

**Issue by sten0** [Sarcasm/irony-mode#587]
Hi @Sarcasm

This test recently started failing in Debian, and it's a release critical bug for our copy of Irony-mode.  Would you please take a look?

https://bugs.debian.org/1052955

```elisp
Package cl is deprecated
Loading /<<PKGBUILDDIR>>/server/test/elisp/test-config.el (source)...
irony-iotask.el: Warning: Case 'value will match ‘quote’.  If that’s intended, write (value quote) instead.  Otherwise, don’t quote ‘value’.
irony-iotask.el: Warning: Case 'error will match ‘quote’.  If that’s intended, write (error quote) instead.  Otherwise, don’t quote ‘error’.
irony.el: Warning: Use keywords rather than deprecated positional arguments to `define-minor-mode'
Test ‘irony/split-command-line/with-quotes’ redefined

Error: error ("Test ‘irony/split-command-line/with-quotes’ redefined")
  mapbacktrace(#f(compiled-function (evald func args flags) #<bytecode 0x23b4b4debae87eb>))
  debug-early-backtrace()
  debug-early(error (error "Test ‘irony/split-command-line/with-quotes’ redefined"))
  error("Test `%s' redefined" irony/split-command-line/with-quotes)
  ert-set-test(irony/split-command-line/with-quotes #s(ert-test :name irony/split-command-line/with-quotes :documentation "Test if files are removed from the arguments list.\n\nhttps://github.com/Sarcasm/irony-mode/issues/101" :body (lambda nil (let ((cmd-line "g++ -DFOO=\\\"\\\"")) (let* ((fn-47 #'equal) (args-48 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list '("g++" "-DFOO=\"\"") (irony--split-command-line cmd-line))) (error (progn (setq fn-47 #'signal) (list (car err) (cdr err))))))) (let ((value-49 'ert-form-evaluation-aborted-50)) (let (form-description-51) (if (unwind-protect (setq value-49 (apply fn-47 args-48)) (setq form-description-51 (nconc (list '(should (equal '("g++" "-DFOO=\"\"") (irony--split-command-line cmd-line)))) (list :form (cons fn-47 args-48)) (if (eql value-49 'ert-form-evaluation-aborted-50) nil (list :value value-49)) (if (eql value-49 'ert-form-evaluation-aborted-50) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-48)) nil))))) (ert--signal-should-execution form-description-51)) nil (ert-fail form-description-51))) value-49)))) :most-recent-result nil :expected-result-type :passed :tags nil :file-name "/<<PKGBUILDDIR>>/server/test/elisp/irony.el"))
  load-with-code-conversion("/<<PKGBUILDDIR>>/server/test/elisp/irony.el" "/<<PKGBUILDDIR>>/server/test/elisp/irony.el" nil t)
  command-line-1(("-l" "package" "--eval" "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa\")" "--eval" "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa-src\")" "-f" "package-initialize" "-L" "." "--eval" "(require 'cl)" "-l" "server/test/elisp/irony.el" "-l" "server/test/elisp/irony-iotask.el" "-l" "server/test/elisp/irony-cdb-json.el" "--eval" "(ert-run-tests-batch-and-exit)"))
  command-line()
  normal-top-level()
```

**Comment by Sarcasm** [comment]
Hello,

Thank you for reporting the issue.
I just pushed a fix for this issue and some other warnings.

I released a new tag v1.6.0 with all the fixes.

I will let you close this ticket if the build pass again on Debian.

Thank you.

**Comment by sten0** [comment]
Guillaume Papin ***@***.***> writes:

> Hello,
>
> Thank you for reporting the issue.

You're welcome :)

> I just pushed a fix for this issue and some other warnings.
>
> I released a new tag v1.6.0 with all the fixes.

Thanks, much appreciated!

> I will let you close this ticket if the build pass again on Debian.

I've confirmed that everything is good on my local amd64 (iirc what
Fedora calls x86_64) machine, so I'll close this momentarily.

**Comment by sten0** [comment]
Closed by 4a50775
